libs/test_positivity.py

```diff
@@ -0,0 +1,142 @@
+import dataclasses
+import pathlib
+from itertools import chain
+from typing import Sequence
+
+import pandas as pd
+
+from covidactnow.datapublic.common_fields import CommonFields, FieldName
+from covidactnow.datapublic.common_fields import PdFields
+
+from libs.datasets.timeseries import MultiRegionTimeseriesDataset
+
+
+@dataclasses.dataclass
+class Method:
+    """A method of calculating test positivity"""
+
+    name: str
+    numerator: FieldName
+    denominator: FieldName
+
+    def calculate(self, delta_df: pd.DataFrame) -> pd.DataFrame:
+        assert delta_df.columns.names == [CommonFields.DATE]
+        assert delta_df.index.names == [PdFields.VARIABLE, CommonFields.LOCATION_ID]
+        # delta_df has the field name as the first level of the index. delta_df.loc[field, :] returns a
+        # DataFrame without the field label so operators such as `/` are calculated for each
+        # region/state and date.
+        return delta_df.loc[self.numerator, :] / delta_df.loc[self.denominator, :]
+
+
+TEST_POSITIVITY_METHODS = (
```

Comment:
These kinda feel like they could an enum

Reply:
Sort of, but I think it'd make testing using a different set of methods harder.